pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -255,8 +253,7 @@ def on_load_checkpoint(self, callback_state: Dict[str, Any]):
     def save_checkpoint(self, trainer, unused: Optional = None):
         """
         Performs the main logic around saving a checkpoint.
-        This method runs on all ranks, it is the responsibility of `self.save_function`
-        to handle correct behaviour in distributed training, i.e., saving only on rank 0.
+        This method runs on all ranks.
```

Comment:
I would not remove this information. It should be good to know that it's not the callback that does the actual save operation.